core/src/main/java/io/questdb/cairo/PlainRecordChain.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo;
+
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.SqlExecutionCircuitBreaker;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryARW;
+import io.questdb.std.MemoryTag;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * PlainRecordChain is similar to RecordChain, except that it stores the record's startOffset in a separate memory
+ * location instead of at the record header.
+ * This enhances its random access capability, making it behave more like an array,
+ * while sacrificing the ability to access records like a linked list.
+ */
+public class PlainRecordChain extends RecordChain {
+
+    private final MemoryARW recordsStartOffset;
+    private long size = 0L;
+    private long nextRecordIndex = 0L;
+
+    public PlainRecordChain(@NotNull ColumnTypes columnTypes, @NotNull RecordSink recordSink, long pageSize, int maxPages) {
+        super(columnTypes, recordSink, pageSize, maxPages);
+        this.recordsStartOffset = Vm.getCARWInstance(pageSize >> 3, maxPages, MemoryTag.NATIVE_RECORD_CHAIN);
```

Comment:
we have to have a lower limit on the page size for the aux memory

Reply:
hey Victor, thank for you the the adjustment. Sorry, to mess you around. I think `pageSize` is a good value. We recently did a change to the underlying memory, which would extend memory in square of the size, which is very fast even for a low starting point. Speaking of starting point  - it is only 1MiB. It is ok to have RecordList with 1MiB initially, we don't need to go any smaller.